modules/ipc/include/hephaestus/ipc/zenoh/service.h

```diff
@@ -4,30 +4,203 @@
+        << "Encoding for binary types should be Z_ENCODING_PREFIX_EMPTY";
+    RequestT request;
+    auto payload = query.get_value().get_payload();
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    std::span<const std::byte> buffer(reinterpret_cast<const std::byte*>(payload.start), payload.len);
+    buffer = internal::removeChangingBytes(buffer);
+    DLOG(INFO) << "Deserializing buffer of size: " << buffer.size();
+    DLOG(INFO) << fmt::format("Payload: {}", fmt::join(buffer, ","));
+    serdes::deserialize<RequestT>(buffer, request);
+    return request;
+  }
+}
+
+template <class ReplyT>
+auto onReply(zenohc::Reply&& reply, std::vector<ServiceResponse<ReplyT>>& reply_messages, std::mutex& m,
+             const std::string& topic_name) -> void {
+  const auto result = std::move(reply).get();
+  if (const auto& error = std::get_if<zenohc::ErrorMessage>(&result)) {
+    LOG(ERROR) << fmt::format("Error on '{}' reply: '{}'", topic_name, error->as_string_view());
+    return;
+  }
+
+  const auto& sample = std::get_if<zenohc::Sample>(&result);
+  DLOG(INFO) << fmt::format("Received answer of '{}'", sample->get_keyexpr().as_string_view());
+  if constexpr (std::is_same_v<ReplyT, std::string>) {
+    CHECK(sample->get_encoding() == Z_ENCODING_PREFIX_TEXT_PLAIN)
+        << "Encoding for std::string should be Z_ENCODING_PREFIX_TEXT_PLAIN";
+    DLOG(INFO) << fmt::format("Payload is string: '{}'", sample->get_payload().as_string_view());
+    std::unique_lock<std::mutex> lock(m);
+    reply_messages.emplace_back(ServiceResponse<ReplyT>{
+        .topic = topic_name, .value = static_cast<std::string>(sample->get_payload().as_string_view()) });
+  } else {
+    CHECK(sample->get_encoding() == Z_ENCODING_PREFIX_EMPTY)
+        << "Encoding for binary types should be Z_ENCODING_PREFIX_EMPTY";
+    auto payload = sample->get_payload();
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    std::span<const std::byte> buffer(reinterpret_cast<const std::byte*>(payload.start), payload.len);
+    ReplyT reply_deserialized;
+    serdes::deserialize(buffer, reply_deserialized);
+    std::unique_lock<std::mutex> lock(m);
+    reply_messages.emplace_back(topic_name, std::move(reply_deserialized));
+  }
+}
+}  // namespace internal
+
+template <typename RequestT, typename ReplyT>
+Service<RequestT, ReplyT>::Service(SessionPtr session, TopicConfig topic_config, Callback&& callback)
+  : session_(std::move(session)), topic_config_(std::move(topic_config)), callback_(std::move(callback)) {
+  internal::checkTemplatedTypes<RequestT, ReplyT>();
+
+  auto query = [this](const zenohc::Query& query) mutable {
+    LOG(INFO) << fmt::format("Received query from '{}'", query.get_keyexpr().as_string_view());
+
+    auto reply = this->callback_(internal::deserializeRequest<RequestT>(query));
+    DLOG(INFO) << fmt::format("Replying to '{}'", query.get_keyexpr().as_string_view());
```

Comment:
you are not printing any new information wrt line 126 so I would remove this line